vortex-array/src/patches.rs

```diff
@@ -122,7 +122,12 @@ impl Patches {
     /// * Indices is an unsigned integer type
     /// * Indices must be sorted
     /// * Last value in indices is smaller than array_len
-    pub fn new_unchecked(array_len: usize, offset: usize, indices: Array, values: Array) -> Self {
+    pub unsafe fn new_unchecked(
```

Comment:
is it actually unsafe memory-wise? I think we should reserve `unsafe` for those cases, and just have very clear docs for correctness concerns.

Reply:
ah yea  sure makes sense. This itself is not memory unsafe now, but would be if we add other unsafe code that relies on the checks that this skips